test/test_cache.py

```diff
@@ -159,16 +162,29 @@ def test_inactive_cache(tmpdir, isolated_parso_cache):
     paths = os.listdir(raw_cache_path)
     a_while_ago = time.time() - _CACHED_FILE_MAXIMUM_SURVIVAL
     old_paths = set()
-    for path in paths[:len(test_subjects) // 2]: # make certain number of paths old
+    for path in paths[:len(test_subjects) // 2]:  # make certain number of paths old
         os.utime(os.path.join(raw_cache_path, path), (a_while_ago, a_while_ago))
         old_paths.add(path)
     # nothing should be cleared while the lock is on
     assert os.path.exists(_get_cache_clear_lock().path)
-    _remove_cache_and_update_lock() # it shouldn't clear anything
+    _remove_cache_and_update_lock()  # it shouldn't clear anything
     assert len(os.listdir(raw_cache_path)) == len(test_subjects)
     assert old_paths.issubset(os.listdir(raw_cache_path))
 
     os.utime(_get_cache_clear_lock().path, (a_while_ago, a_while_ago))
     _remove_cache_and_update_lock()
     assert len(os.listdir(raw_cache_path)) == len(test_subjects) // 2
     assert not old_paths.intersection(os.listdir(raw_cache_path))
+
+
+@skip_pypy
+def test_permission_error(monkeypatch, recwarn):
+    def save(*args, **kwargs):
+        was_called[0] = True  # Python 2... Use nonlocal instead
+        raise PermissionError
+
+    was_called = [False]
+
+    monkeypatch.setattr(cache, '_save_to_file_system', save)
+    parse(path=__file__, cache=True, diff_cache=True)
+    assert was_called[0]
```

Comment:
Would you also add something to verify the warning has been raised. Something like `pytest.warns(Warning)` or `len(recwarn) > 0`

Reply:
Very good point. I even had `recwarn` in the function signature, but then forgot to implement it. Thanks for mentioning it!